src/services/bundleAnalysis/useBundleSummary.tsx

```diff
@@ -0,0 +1,190 @@
+        })
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repository: z
+        .discriminatedUnion('__typename', [
+          RepositorySchema,
+          RepoNotFoundErrorSchema,
+          RepoOwnerNotActivatedErrorSchema,
+        ])
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const query = `
+query BundleSummary(
+  $owner: String!
+  $repo: String!
+  $branch: String!
+  $bundle: String!
+) {
+  owner(username: $owner) {
+    repository(name: $repo) {
+      __typename
+      ... on Repository {
+        branch(name: $branch) {
+          head {
+            bundleAnalysisReport {
+              __typename
+              ... on BundleAnalysisReport {
+                bundle(name: $bundle) {
+                  name
+                  moduleCount
+                  sizeTotal
+                  bundleData {
+                    loadTime {
+                      threeG
+                      highSpeed
+                    }
+                    size {
+                      gzip
+                      uncompress
+                    }
+                  }
+                }
+              }
+              ... on MissingHeadReport {
+                message
+              }
+            }
+          }
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+interface UseBundleSummaryArgs {
+  provider: string
+  owner: string
+  repo: string
+  branch: string
+  bundle: string
+}
+
+export const useBundleSummary = ({
+  provider,
+  owner,
+  repo,
+  branch,
+  bundle,
+}: UseBundleSummaryArgs) => {
+  return useQuery({
+    queryKey: ['BundleSummary', provider, owner, repo, branch, bundle],
+    queryFn: ({ signal }) =>
+      Api.graphql({
+        provider,
+        query,
+        signal,
+        variables: { owner, repo, branch, bundle },
+      }).then((res) => {
+        const parsedData = RequestSchema.safeParse(res?.data)
+
+        if (!parsedData.success) {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useBundleSummary - 404 Failed to parse data',
+          } satisfies NetworkErrorObject)
+        }
+
+        const data = parsedData.data
+
+        if (data?.owner?.repository?.__typename === 'NotFoundError') {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useBundleSummary - 404 Not found error',
+          } satisfies NetworkErrorObject)
+        }
+
+        if (data?.owner?.repository?.__typename === 'OwnerNotActivatedError') {
+          return Promise.reject({
+            status: 403,
+            data: {
+              detail: (
+                <p>
+                  Activation is required to view this repo, please{' '}
+                  {/* @ts-expect-error */}
```

Comment:
curious about this ts-expect-error

Reply:
This is just a copy paste, ignoring the type issues with the A tag

Reply:
Fair enough, plan to swap to applink in a future PR?

Reply:
No we don't use the AppLink for in components directly for links like this, the `A` is built on top of it for that reason

Reply:
oh right, brain fart there lol